glean-core/src/upload/mod.rs

```diff
@@ -0,0 +1,225 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Manages the pending pings queue and directory.
+//!
+//! * Keeps track of pending pings, loading any unsent ping from disk on startup;
+//! * Exposes `get_next_ping` API for the platform layer to request next ping in line;
+//! * Exposes `process_ping_upload_response` API to check the HTTP response from the ping upload
+//!   and either delete the corresponding ping from disk or re-enqueue it for sending.
+
+// !IMPORTANT!
+// Remove the next line when this module's functionality is in the Glean object.
+// This is here just to not have lint error for now.
+#![allow(dead_code)]
```

Comment:
Isn't this already in the Glean object?

Reply:
It was, but since it is not anymore I'll leave this :)